### ISSUE TITLE
ph with keyref is not converted to link in PDF output [DOT 2.x develop branch]

### ISSUE THREAD

**Issue by raducoravu** [dita-ot/dita-ot#2025]
If my DITA Map has a key definition like:

```
    <topicref keys="CONTAINER" href="topics/care.dita">
      <topicmeta>
        <linktext>CONTAINER</linktext>
        <keywords>
          <keyword>CONTAINER</keyword>
        </keywords>
      </topicmeta>
    </topicref>
```

and another DITA topic has a keyref like:

```
<ph keyref="CONTAINER"/>
```

in the XHTML output the keyref becomes a link. But in the PDF output it does not.

**Comment by robander** [comment]
Think this is related to some other issue where we've discussed having the key resolution module add some namespaced attributes to hold the linking information, when used on a non-linking element. Of course that is not quite as simple as it sounds...

**Comment by jelovirt** [comment]
I've already changed the code so that `<ph>` will get the attributes in a namespace. Support for that is probably not ported to PDF.

**Comment by robander** [comment]
Thought so just couldn't remember if it was already there or if there was still an open issue. Should have known that if we chatted about it a while ago, you'd already have it done (at least that keyref module update).

**Comment by alastrina123** [comment]
The output @radcoravu expects is achieved with the xref element. I don't think you should apply this behaviour to the ph element

**Comment by raducoravu** [comment]
@alastrina123 The DITA 1.3 specs:

https://www.oxygenxml.com/dita/1.3/specs/archSpec/base/processing-keyref-for-links.html


         When a key definition is bound to a resource that is addressed by the @href or @keyref attributes,       
          and does not specify "none" for the @linking attribute, all references to that key definition become links to the bound resource. 
         When a key definition is not bound to a resource or specifies "none" for the @linking attribute, references to that key definition do not become links.

**Comment by alastrina123** [comment]
@raducoravu apologies I don't have your level of familiarity with the spec... but, if I set @linking to none, what output would an xref element generate?

**Comment by raducoravu** [comment]
I would assume the xref no longer would translate to a link, it becomes plain text, similar to a ph with keyref pointing to the same key.

**Comment by alastrina123** [comment]
Ok... we organise many keydefs in container maps, each with href and keyword info. We then call the href with the xref element and the keyword content with the ph element. If I want the keyword content to form a link I do this

**Comment by alastrina123** [comment]
Continued from above... I nest the ph element in the xref element

**Comment by robander** [comment]
Following the specification -- I don't think there is a defined way to create a key definition such that:

- Some references to that key result in a link + link text, but
- Other references to that key result in only the text

The assumption when creating this mechanism was that if a URI is present on the key definition, it should be used. This has been working for a long time in XHTML / HTML (both `xref` and `ph` result in a link), but PDF was out of sync. 

The only ways I can think of to do this are to have two key definitions (one with `@href`, one without) or to put an ID on the link text, and when you want to reuse that text without a link, refer to it with `@conref` instead of with `@keyref`. I realize neither of those options are really great though (which probably comes from the fact that this use case wasn't previously discussed by the OASIS standards group).

**Comment by raducoravu** [comment]
@robander One of those cases in which the publishing engine works in slightly different way than specs and somehow users start using certain constructs based on what works in the current output.
In this case maybe reusing the linktext between two key definitions might have also worked:

          <keydef keys="k_PlainText">
            <topicmeta>
              <linktext conref="#lt"/>
            </topicmeta>
          </keydef>
          <keydef keys="k_Link" href="http://www.google.com" scope="external" format="html">
            <topicmeta>
             <linktext id="lt">Some link text</linktext>
            </topicmeta>
          </keydef>

but not sure if it works because of https://github.com/dita-ot/dita-ot/issues/2420

**Comment by robander** [comment]
The issue is fixed with #2961; PDF will now be in sync with HTML and XHTML for inline elements with `@keyref`.

**Comment by raducoravu** [comment]
Great, thanks.
